### PR TITLE
guards for subdust in opreturn

### LINKED ISSUE #841: Ensure subdust guards
* Do not allow op returns with value if they are not in the sub dust format
* Do not allow op return value >= subdust

### PR DESCRIPTION
**Pull request by bitcoin-coder-bob**
Closes #841

*To be rebased off of https://github.com/arkade-os/arkd/pull/923 after it is merged first*

Created a functions `validateOffchainTxOutputs` called from within existing `SubmitOffchainTx` to separate out the validation logic. Within this logic is the addition of the 2 new guards outlined in the issue.

Tests were added against this new function in the new test file: `internal/core/application/validate_outputs_test.go`

## Summary by CodeRabbit

* **Refactor**
  * Centralized and streamlined offchain transaction output validation, consolidating checks for anchors, OP_RETURN, dust, and per-output value limits while preserving existing behavior.

* **Tests**
  * Added comprehensive tests covering valid flows, error conditions, boundary values, and mixed scenarios to reduce regressions.

### CODE REVIEW

File: internal/core/application/validate_outputs_test.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+				anchor,
+				{Value: 600, PkScript: testP2TRScript(t)},
+			},
+			dust:                    testDust,
+			vtxoMaxAmount:           -1,
+			vtxoMinOffchainTxAmount: 1000,
+			wantErr:                 true,
+			wantErrCode:             errors.AMOUNT_TOO_LOW.Code,
+			wantErrContains:         "lower than min vtxo amount",
+		},
```
[the remark this answers is not in the document]
Comment (bitcoin-coder-bob) [reply internal/core/application/validate_outputs_test.go]: done in: [ab31d0d](https://github.com/arkade-os/arkd/pull/925/commits/ab31d0d306d6930c5ccf38595983ab14b03bf5b6)

File: internal/core/application/service.go [hunk 2]
```diff
@@ -4138,6 +4072,115 @@ func (s *service) GetIntentByTxid(
...
+		}
+		if out.Value < vtxoMinOffchainTxAmount {
+			return nil, errors.AMOUNT_TOO_LOW.New(
+				"output #%d amount is lower than min vtxo amount: %d",
+				outIndex, vtxoMinOffchainTxAmount,
+			).WithMetadata(errors.AmountTooLowMetadata{
+				OutputIndex: outIndex,
+				Amount:      int(vtxoMinOffchainTxAmount),
+				MinAmount:   int(vtxoMinOffchainTxAmount),
+			})
```
[the remark this answers is not in the document]
Comment (bitcoin-coder-bob) [reply internal/core/application/service.go:4359]: done in: [ab31d0d](https://github.com/arkade-os/arkd/pull/925/commits/ab31d0d306d6930c5ccf38595983ab14b03bf5b6)

File: internal/core/application/service.go [hunk 3]
```diff
@@ -4278,6 +4194,132 @@ func (s *service) GetIntentByTxid(
...
+			}
+
+			// not subdust format but has value in invalid
+			if out.Value > 0 {
+				return nil, nil, errors.MALFORMED_ARK_TX.New(
+					"OP_RETURN output #%d has non-zero value (%d) but is not a subdust output",
+					outIndex, out.Value,
+				).WithMetadata(errors.PsbtMetadata{Tx: signedArkTx})
+			}
+			outputs = append(outputs, out)
```
[the remark this answers is not in the document]
Comment (louisinger) [reply internal/core/application/service.go]: @bitcoin-coder-bob the rabbit is right, for extension let's check value = 0, fail otherwise (it may change later if we include subdust to extension output but in the meantime better to be explicit). We should also add this check to RegisterIntent, cause the intent may contain extension

File: internal/core/application/validate_outputs_test.go [hunk 4]
```diff
@@ -0,0 +1,428 @@
+package application
```
Comment (louisinger) [inline internal/core/application/validate_outputs_test.go]: let's move this test to service_test.go

**Review by louisinger — CHANGES_REQUESTED** [review]

File: internal/core/application/validate_outputs_test.go (hunk 4 above)
Comment (bitcoin-coder-bob) [reply internal/core/application/validate_outputs_test.go]: done in: [1b79621](https://github.com/arkade-os/arkd/pull/925/commits/1b79621800edb77e9533cf2d8cbfba2a6eae3d99)

File: internal/core/application/service.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bitcoin-coder-bob) [reply internal/core/application/service.go]: done in: [1b79621](https://github.com/arkade-os/arkd/pull/925/commits/1b79621800edb77e9533cf2d8cbfba2a6eae3d99)
